sign/impl.go

```diff
@@ -16,14 +16,39 @@ limitations under the License.
 
 package sign
 
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"github.com/sigstore/cosign/cmd/cosign/cli/options"
+	"github.com/sigstore/cosign/cmd/cosign/cli/sign"
+)
+
 type defaultImpl struct{}
 
 //go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 -generate
 //counterfeiter:generate . impl
+//go:generate /usr/bin/env bash -c "cat ../scripts/boilerplate/boilerplate.generatego.txt signfakes/fake_impl.go > signfakes/_fake_impl.go && mv signfakes/_fake_impl.go signfakes/fake_impl.go"
 type impl interface {
 	VerifyInternal(*Signer, string) (*SignedObject, error)
+	SignImageInternal(ctx context.Context, ko sign.KeyOpts, regOpts options.RegistryOptions,
+		annotations map[string]interface{}, imgs []string, certPath string, upload bool,
+		outputSignature string, outputCertificate string, payloadPath string, force bool,
+		recursive bool, attachment string) error
 }
 
 func (*defaultImpl) VerifyInternal(signer *Signer, reference string) (*SignedObject, error) {
 	return signer.Verify(reference)
 }
+
+func (*defaultImpl) SignImageInternal(ctx context.Context, ko sign.KeyOpts, regOpts options.RegistryOptions, // nolint: gocritic
+	annotations map[string]interface{}, imgs []string, certPath string, upload bool,
+	outputSignature string, outputCertificate string, payloadPath string, force bool,
+	recursive bool, attachment string) error {
+	err := sign.SignCmd(context.Background(), ko, regOpts, annotations, imgs, certPath, upload, outputSignature, outputCertificate, payloadPath, force, recursive, attachment)
```

Comment:
Can we use the cosign package library (`github.com/sigstore/cosign/pkg`) instead of wrapping the CLI (`github.com/sigstore/cosign/cmd/cosign/cli`)?

Reply:
will check

Reply:
it does a lot of things that if we use the pkg we will need to do in our side

Reply:
maybe that part should be in the pkg, but it is the way it was implemented

Reply:
Yeah, a lot of the code is currently hidden in https://github.com/sigstore/cosign/tree/main/internal/pkg/cosign
From what I saw doing it like @cpanato did or reimplementing a bunch of code like I was doing here: https://github.com/puerco/release-sdk/blob/enable-signer/sign/impl.go

I am happy if we do it this way for now so that we can start using the library and when the libraries above are exposed we import them.

Reply:
agree with you, when we have the libs exported in a clean way we can refactor